src/celeritas/ext/VecgeomParams.cc

```diff
@@ -181,6 +181,22 @@ void VecgeomParams::build_tracking()
 #if CELERITAS_USE_CUDA
         auto& cuda_manager = vecgeom::cxx::CudaManager::Instance();
 
+        // Insure that kernels, for which the compiler is not able to determine
+        // the total stack usage, have enough stack reserved.
+        // See https://github.com/celeritas-project/celeritas/issues/614 for
+        // a more detailed discussion.
+        // Experimentally, this seems to be needed only in the following cases.
+        // (a stricter condition could have been
+        //   if use_vecgeom && (compiled without -O2 || CELERITAS_DEBUG)
+        //      || (CELERITAS_DEBUG && compiled with -g)
+        // with the 2nd part being usefull only if vulnerable kernels start
+        // appearing in the build; for this second part we ought to move (or
+        // more exactly copy) to `celeritas::activate_device` as the latest
+        // cudaDeviceSetLimit 'wins'.
+    #if (CELERITAS_USE_VECGEOM && CELERITAS_DEBUG)
+        set_cuda_stack_size(16384);
+    #endif
```

Comment:
I like that this was set in VecgeomParams instead of in the other places I suggested: much better.

- Since this is in VecgeomParams we can leave out `CELERITAS_USE_VECGEOM` 😄 
- Since the function is defined whether or not CUDA is enabled  we should use `if (MACRO)` to catch any misspellings and because C++ is cleaner than preprocessor code

```suggestion
        if (CELERITAS_DEBUG)
        {
            set_cuda_stack_size(16384);
        }
```

Reply:
> I like that this was set in VecgeomParams instead of in the other places I suggested:

Technically we may have to move (well copy) it to `celeritas::activate_device` if (when?) we have a test that fails (in the VecGeom build) without even initializing VecGeom.  For now it works as is so good enough.